datastore/importers/nmdb-import-systeminfo/Parser.cpp

```diff
@@ -0,0 +1,238 @@
+
+    networkCard =
+        ('[' >> qi::lexeme[+qi::char_("0-9")] >> ']' >> qi::lit(':') 
+        >> networkCardName[(pnx::bind(&Parser::addInterface, this, qi::_1))] 
+        >> qi::eol 
+        >> "Connection Name: " 
+        > networkCardConnectionName[(pnx::bind(&Parser::addIfaceConnectName, this, qi::_1))] 
+        >> qi::eol 
+        >> -(dhcpEnabledStatus 
+        >> qi::eol) 
+        >> -("DHCP Server: " 
+        >> dhcpServer >> qi::eol) 
+        >> -("IP address(es)" 
+        >> qi::eol 
+        > +('[' >> qi::lexeme[+qi::char_("0-9")] >> "]: "
+        >> ipAddr[(pnx::bind(&Parser::addIfaceIp, this, qi::_1))] >> qi::eol)) 
+        >> -("Status: " 
+            >> networkCardStatus[(pnx::bind(&Parser::setIfaceStatus, this, qi::_1))] 
+            >> qi::eol)
+        );
+    ;
+    networkCards =
+        qi::lit("Network Card(s):") 
+        >> +qi::ascii::print 
+        > qi::eol 
+        > +networkCard;
+
+    systemInfo =
+        hostName 
+        > osName
+        > osVersion 
+        > osManufacturer 
+        > osConfiguration 
+        > *(qi::char_ - qi::lit("System Manufacturer:")) //Skip to system manufacturer
+        > systemManufacturer 
+        > systemModel 
+        > systemType 
+        > *(qi::char_ - qi::lit("Domain:")) //Skip to domain
+        > -domain[pnx::ref(curDomain) = qi::_1] 
+        > *(qi::char_ - qi::lit("Hotfix")) //Skip to hotfixes
+        > -hotfixes 
+        > -networkCards 
+        > qi::omit[*(qi::char_ - qi::eoi)]; //omit the rest of sysinfo (aka hyper-v)
+    ;
+    ignoredLine =
+        (qi::ascii::print > -qi::eol) | +qi::eol;
+
+    // Allows for error handling and debugging of qi.
+    BOOST_SPIRIT_DEBUG_NODES(
+        (start)
+        (systemInfo)
+        (hostName)
+        (osName)
+        (osVersion)
+        (osManufacturer)
+        (osConfiguration)
+        (systemManufacturer)
+        (systemModel)
+        (systemType)
+        (domain)
+        (hotfixes)
+        (hotfix)
+        (networkCards)
+        (networkCard)
+        (dhcpServer)
+        (dhcpEnabledStatus)
+        (networkCardName)
+        (networkCardConnectionName)
+        (networkCardStatus));
+}
+
+// =============================================================================
+// Parser helper methods
+// =============================================================================
+
+void Parser::addHotfix(const std::string &_string)
+{
+    data.hotfixes.push_back(_string);
+}
+
+void Parser::addInterface(const std::string &_name)
+{
+    nmdo::Interface iface;
+    iface.setName(_name);
+    curIfaceName = iface.getName();
+    data.network_cards[curIfaceName] = iface;
+}
+
+void Parser::addIfaceConnectName(const std::string &_connectionname)
+{
+    auto &iface{data.network_cards[curIfaceName]};
+    if (_connectionname.find("Wi-Fi") != std::string::npos || _connectionname.find("Bluetooth") != std::string::npos)
+    {
+        iface.setMediaType(_connectionname);
+    }
```

Comment:
Wait this is weird, how is the mediaType set to ethernet then if I only have the set function in this conditional? It actually doesn't.... why?? no clue...
[name: intel(r) wi-fi 6 ax200 160mhz, description: Wi-Fi, isUp: false, mediaType: wi-fi, macAddr: [macAddress: Invalid MAC, ipAddrs: [], isResponding: false]]

Reply:
For `InterfaceNetwork`, the `mediaType` variable is defaulted to `ethernet`.  The problem I'm talking about is if one of "Wi-Fi" or "Bluetooth" is present with more text it will set `mediaType` to the whole text.  The value of `_connectionname` is the rest of the line after `Connection Name:` and that remainder can be several words long.

Reply:
Right, But look at the example I pasted in the previous comment? The actual interface was intel(r) wi-fi 6 ax200 160mhz, but the media-type got set to wi-fi

Reply:
Okay I just added some if statements to properly set the media type